Source/Particles/Collision/BinaryCollision/BinaryCollision.H

```diff
@@ -224,7 +247,7 @@ public:
         using namespace ParticleUtils;
         using namespace amrex::literals;
 
-        CollisionFunctorType binary_collision_functor = m_binary_collision_functor;
+        CollisionFunctor binary_collision_functor = m_binary_collision_functor;
```

Comment:
first thought: We could use an alias here (or a pointer)
```suggestion
        CollisionFunctor & binary_collision_functor = m_binary_collision_functor;
```

Reply:
Somehow using a reference wasn't enough and I still got compiler complaints about trying to copy `ScatteringProcess` instances. I was able to work around this issue by only keeping the `ScatteringProcess::Executor` structs instead of the entire class but it requires that I move the cross-section data from each `ScatteringProcess` class to the `DSMCFunc` instance.

Reply:
This issue is solved by using the internal `struct` approach 🎉